### ISSUE TITLE
How long does it take to run this program?

### ISSUE THREAD

**Issue by zoltar9264** [hidasib/GRU4Rec#17]
Hello, I am a beginner of python and Neural Network 。 I run this code in a 32 cores cpu without GPU.
"run_rsc15.py" has run amost 24 hours. So I want know how long it will take to run "python run_rsc15.py". by the way, memory of the server is 128G.
think you very much !

**Comment by zoltar9264** [comment]
finally,this program finished with 51 hours.

**Comment by hidasib** [comment]
Please note that training is optimized or GPU and not CPU. On GPU training should finish under 6 hours (depending on the parameters you use it can be much faster). If you plan to stik to CPU I'd suggest using the cross-entropy loss as it uses less of the GPU specific speed-ups that make the network train slower on CPU. (Note that opimal hyperparameters for the cross-entropy loss are different than for BPR-max.)

Additional speed-up for GPU will be published shortly.

**Comment by CrisMon-01** [comment]
@hidasib 
How can I see if it's running on GPU?
I have run the gpu_ops.py rogram.
After that i hae run the run_rsc15.py
but i see on my machine (ubuntu with a nvidia 960M) with the command:
nvidia-smi -l 
there is no python programm runnign n that
